src/shared/clients/devenvClient.ts

```diff
@@ -3,14 +3,65 @@
  * SPDX-License-Identifier: Apache-2.0
  */
 
+import * as vscode from 'vscode'
 import got from 'got'
+import globals from '../extensionGlobals'
+import { getLogger } from '../logger/logger'
 import { getCodeCatalystDevEnvId } from '../vscode/env'
 
 const environmentAuthToken = '__MDE_ENV_API_AUTHORIZATION_TOKEN'
 const environmentEndpoint = 'http://127.0.0.1:1339'
 
-export class DevEnvClient {
-    public constructor(private readonly endpoint: string = environmentEndpoint) {}
+/**
+ * Client to the MDE quasi-IMDS localhost endpoint.
+ */
+export class DevEnvClient implements vscode.Disposable {
+    static #instance: DevEnvClient
+    private readonly timer
+    private lastStatus = ''
+    private onStatusChangeFn: undefined | ((oldStatus: string, newStatus: string) => void)
+
+    /** Singleton instance (to avoid multiple polling workers). */
+    public static get instance() {
+        return (this.#instance ??= new this())
+    }
+
+    /** @internal */
+    public constructor(private readonly endpoint: string = environmentEndpoint) {
+        if (!this.id) {
+            getLogger().debug('codecatalyst: DevEnvClient skipped (local)')
+            this.timer = undefined
+        } else {
+            getLogger().info('codecatalyst: DevEnvClient started')
+            this.timer = globals.clock.setInterval(async () => {
+                const r = await this.getStatus()
+                if (this.lastStatus !== r.status) {
+                    const newStatus = r.status ?? 'NULL'
+                    getLogger().info(
+                        'codecatalyst: DevEnvClient: status change (old=%s new=%s) action=%s: %s',
```

Comment:
Currently this just logs status changes. Later could be used to handle `IMAGES-UPDATE-AVAILABLE`.